lib/pure/parsexml.nim

```diff
@@ -669,6 +672,21 @@ proc parseAttribute(my: var XmlParser) =
             pendingSpace = false
           add(my.b, buf[pos])
           inc(pos)
+  elif allowUnquotedAttribs in my.options:
+    const disallowedChars = {'"', '\'', '`', '=', '<', '>', ' ',
+                             '\0', '\t', '\L', '\F', '\f'}
+    let startPos = pos
+    while (let c = buf[pos]; c notin disallowedChars):
```

Comment:
You should probably protect against an IndexError here.

Reply:
Nah, it's consistent with the rest of the code which just checks for `'\0'`. Nothing should `inc` past EOF.